content/blog/pulumi-azure-2-0/_index.md

```diff
@@ -0,0 +1,149 @@
+
+Here is a snippet that defines an Ubuntu VM:
+
+``` ts
+const vm = new azure.compute.LinuxVirtualMachine("server-vm", {
+    resourceGroupName,
+    networkInterfaceIds: [networkInterface.id],
+    size: "Standard_A0",
+    sourceImageReference: {
+        publisher: "Canonical",
+        offer: "UbuntuServer",
+        sku: "16.04-LTS",
+        version: "latest",
+    },
+    osDisk: {
+        caching: "ReadWrite",
+        storageAccountType: "Standard_LRS",
+    },
+    computerName: "hostname",
+    adminUsername: username,
+    adminPassword: password,
+    disablePasswordAuthentication: false,
+);
+```
+
+## Removing Deprecated Resources, Invokes, and Fields
+
+Azure Active Directory has its [own Pulumi provider](https://github.com/pulumi/pulumi-azuread/), so all the AD resources were now removed from the `AD` namespace of the Azure provider 2.0.
+
+A bunch of other resources, invokes, and fields were removed too, following the changes in the upstream provider. You can see the full list in [this upgrade guide](https://www.terraform.io/docs/providers/azurerm/guides/2.0-upgrade-guide.html).
+
+## Default Versions in Callback Functions
+
+[Serverless as Callbacks](https://www.pulumi.com/blog/serverless-as-simple-callbacks-with-pulumi-and-azure-functions/) running on Azure Functions have bumped the default version of Azure Functions runtime to `~3` and `Node.js` to `~12`. As always, you can override the default to set the versions that you require.
+
+Callback Functions also moved from `ZipBlob` to `Blob` deploy as described above, so expect another replacement operation while upgrading.
+
+## Custom Timeouts
+
+Some types of resources take longer to create than others. For instance, provisioning a new Cosmos DB account may take from 10 minutes to over an hour depending on the number of geo locations and other factors.
+
+If you find that creation of a target resource times out, you can override the default timeout using resource options:
+
+``` ts
+const cosmosdbAccount = new azure.cosmosdb.Account("cosmosdb", {
+    resourceGroupName: resourceGroup.name,
+    offerType: "Standard",
+    geoLocations: manyLocations,
+    consistencyPolicy: { consistencyLevel: "Session" },
+}, {
+    customTimeouts: {
+        create: "30m",
+        update: "60m",
+    },
+});
+```
+
+You shouldn't need this ability often, but it can be a lifesaver when you do.
+
+## Getting Started with Migration
+
+To get started with Azure provider 2.0, update the versions in your package manager:
+
+```
+// Node.js: package.json
+"@pulumi/azure": "^2.0.0",
+
+// .NET: csproj/fsproj/vbproj
+<PackageReference Include="Pulumi.Azure" Version="2.0.0-preview" />
+
+// Python: requirements.txt
+pulumi-azure>=2.0.0
+
+// Go: Gopkg.toml
+[constraint]]
+  version = "v2.0.0"
+  name = "github.com/pulumi/pulumi-azure"
+```
+
+There are quite a few breaking changes, so you may need to adjust your code before you would be able to run the program successfully again.
```

Comment:
Anything notable we ought to call out? Or maybe link again to the TF upgrade guide?

Reply:
Added another link to the upgrade guide. Does it look better now?